Hub XMLRPC API clarifications (#457)
uyuni-project/uyuni-docs: modules/large-deployments/pages/hub-install.adoc

[[lsd-hub-install]]
= Hub Installation

Before you begin, you need to install the ``hub-xmlrpc-api`` package, and configure the Hub Server to use the API.

.Procedure: Installing and Configuring the Hub XMLRPC API
. On the Hub Server, or on a host that has access to all peripheral servers' XMLRPC APIs, install the ``hub-xmlrpc-api`` package.
The package is available in the {productname} {productnumber} repositories.

> . OPTIONAL: the Hub XMLRPC API service does not start on installation. You can set it up to start automatically with systemd, by executing the following commands

Reviewer: ```suggestion
. OPTIONAL: Set the Hub XMLRPC API service to start automatically on installation:
+
```

Author: Hmm, the intended message is "with the commands below the service will start on each boot (`enable`) and now (`start`)" - merely installing the rpm package gives you neither.

Would the following wording work?

```suggestion
. OPTIONAL: Set the Hub XMLRPC API service to start automatically at boot time, and start it immediately:
```

Reviewer: sure :)

---
sudo systemctl enable hub-xmlrpc-api.service
sudo systemctl start hub-xmlrpc-api.service
---

> . OPTIONAL: verify contents of the configuration file at  ``/etc/hub/hub.conf``

Reviewer: ```suggestion
. OPTIONAL: Check that these parameters in the [path]``/etc/hub/hub.conf`` configuration file are correct:
```

>   - ``HUB_API_URL``: URL to the Hub Server XMLRPC API endpoint. The default is correct if you are installing `hub-xmlrpc-api` on the Hub Server
>   - ``HUB_CONNECT_TIMEOUT``: maximum number of seconds to wait for a response when connecting to a Server. In most cases, these parameter can be left at its default value.
>   - ``HUB_REQUEST_TIMEOUT``: maximum number of seconds to wait for a response when calling a Server method. In most cases, these parameter can be left at its default value.
>   - ``HUB_CONNECT_USING_SSL``: use https instead of plain http for communicating with peripheral Servers. Recommended for a secure environment.

Reviewer: ```suggestion
*  ``HUB_API_URL``: URL to the Hub Server XMLRPC API endpoint. 
Use the default value if you are installing `hub-xmlrpc-api` on the Hub Server.
* ``HUB_CONNECT_TIMEOUT``: the maximum number of seconds to wait for a response when connecting to a Server. 
Use the default value in most cases.
* ``HUB_REQUEST_TIMEOUT``: the maximum number of seconds to wait for a response when calling a Server method. 
Use the default value in most cases.
* ``HUB_CONNECT_USING_SSL``: use HTTPS instead of HTTP for communicating with peripheral Servers. 
Recommended for a secure environment.
```

> [NOTE]
> ====
> In order to use https to connect to peripheral Servers, in addition to setting ``HUB_CONNECT_USING_SSL`` flag to true, SSL certificates for all the peripheral Servers need to be installed on the machine where the hub-xmlrpc-api service runs. This can be achieved by copying the ``RHN-ORG-TRUSTED-SSL-CERT`` certificate file from each peripheral Server's pub directory (http://<server-url>/pub/) to /etc/pki/trust/anchors/ and then running the ``update-ca-certificates`` command.
> ====
> 
> Restart services to pick up any configuration changes.

Reviewer: ```suggestion
. Restart services to pick up configuration changes.

[NOTE]
====
To use HTTPS to connect to peripheral Servers, you must set the ``HUB_CONNECT_USING_SSL`` parameter to ``TRUE``, and ensure that the SSL certificates for all the peripheral Servers are installed on the machine where the ``hub-xmlrpc-api`` service runs. 
Do this by copying the ``RHN-ORG-TRUSTED-SSL-CERT`` certificate file from each peripheral Server's [path]``http://<server-url>/pub/`` directory to [path]``/etc/pki/trust/anchors/``, and run [command]``update-ca-certificates``.
====

```
